server/src/users/users.controller.spec.ts

```diff
@@ -1,23 +1,79 @@
 import { Test, TestingModule } from '@nestjs/testing';
-
+import { userCreateDtoStub, userEntityStub } from '../../test/stubs';
 import { UsersController } from './users.controller';
 import { UsersService } from './users.service';
+import { UsersService as UsersServiceMock } from '../test/__mocks__';
+import { User } from './entities/user.entity';
 
 describe('UsersController', () => {
   let controller: UsersController;
+  let usersService: UsersService = new UsersServiceMock();
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
       controllers: [UsersController],
-      //TODO I put a empty mock for UserService. For real testing we'll need
-      // to create a mock since we are using TypeORM
-      providers: [{ provide: UsersService, useValue: {} }],
+      providers: [{ provide: UsersService, useValue: usersService }],
     }).compile();
 
     controller = module.get<UsersController>(UsersController);
+    usersService = module.get<UsersService>(UsersService);
+    jest.clearAllMocks();
   });
 
   it('should be defined', () => {
     expect(controller).toBeDefined();
   });
+
+  describe('create user', () => {
+    let createdUser: Omit<User, 'password' | 'accept_terms'>;
+
+    describe('when dto is valid', () => {
+      beforeEach(async () => {
+        createdUser = (await controller.create(userCreateDtoStub())) as
+          | Omit<User, 'password' | 'accept_terms'>
+          | any;
```

Comment:
you shouldn't need this `as` 🤔 
also, we def don't want an `any` here

Reply:
because user controller return val
```
  async create(
    @Body() createUserDto: CreateUserDto,
  ): Promise<Omit<User, 'password' | 'accept_terms'>> {
```

Reply:
closing because no longer relevant